pkg/apis/sources/v1alpha1/apiserver_types.go

```diff
@@ -40,6 +40,22 @@ type ApiServerSource struct {
 // Check that we can create OwnerReferences to a Configuration.
 var _ kmeta.OwnerRefable = (*ApiServerSource)(nil)
 
+const (
```

Comment:
I am not sure this should go into the reconciler. this seems real weird because these types are dependent on the adapter that is being used which is independent of the reconciler.

Reply:
talked about this. Agreed that for the current release we will go with this solution of creating the EventTypes in the reconciler, until we figure out a better way, which might involve changing the importers/emitters to expose what types they can send.